ambry-store/src/main/java/com/github/ambry/store/BlobStore.java

```diff
@@ -260,7 +260,9 @@ public void start() throws StoreException {
               config.storeEnableBucketForLogSegmentReports, time, longLivedTaskScheduler, taskScheduler,
               diskIOScheduler, metrics);
         }
-        metrics.initializeIndexGauges(storeId, index, capacityInBytes, blobStoreStats);
+        if (config.storeEnablePerStoreGauges) {
+          metrics.initializeIndexGauges(storeId, index, capacityInBytes, blobStoreStats);
+        }
```

Comment:
The config here is fine but I would consider minimizing its scope. I feel like you can move config into `initializeIndexGauges()` method and only control metric registration of `Gauge<Long> currentInvalidDataSize`.